enhancements/samples-generation.md

```diff
@@ -0,0 +1,613 @@
+- I am a maintainer/contributor, I want to check the result of the changes made in the scaffold by verifying the samples changes , so that I do not need to pull the pull request locally and run the steps to review the code. 
+- I am a maintainer/contributor, I want to have the samples update with the code changes, so that I do not update the samples project after the release
+- I am a maintainer/contributor, I want to have the samples update with the code changes, so that I do not update the samples project after the release
+- I am an operator developer, I want check the examples of projects and features provided, so that I can easily know how to build my operator projects
+- I am an operator developer, I want to check the final result of the tutorial steps, so that I can easily check if what I miss when the things do not go well as expected
+- I am an operator developer, I want to check the samples aligned with the tags release, so that I can check them accordingly to the version. 
+- I am an operator developer, I want to check the samples aligned with the tags release, so that I can easily compare the difference between the releases.
+- I am an operator developer, I want to check the samples aligned with the tags release, so that I can update easily my projects to use an upper version.
+
+#### Re-using samples on tests
+
+- I am a maintainer/contributor, I want to be able to invoke the sample implementation in the code of the e2e tests, so that I do not need to keep maintained the both places.
+- I am an operator developer, I want to be able to check the samples working and running, so that I can be confident that it is a good example for my code implementations. 
+
+#### Helpers and utilities available for tests and samples
+
+- I am a maintainer/contributor, I want to be able to use the helpers in both places, so that I can centralize the code and spend less effort to apply the changes when required 
+
+#### Specific implementation following the same approach 
+
+- I am a maintainer/contributor, I want to feel familiarized with the code implementation, so that I can easily contribute to both places
+- I am a maintainer/contributor, I want the samples and tests implementations are compatible, so that I can easily re-use them and create helpers which are valid for both scenarios
+- I am a maintainer/contributor, I want to have the flexibility to create samples decoupled from the tests, so that I can easily mock scenarios that will not be used in the e2e tests
+- I am a maintainer/contributor, I want to have the flexibility to create samples decoupled from the tests, so that I can easily mock scenarios that will not be used in the e2e tests if required since its reusability concept will be not hurt.
+
+### Implementation Details/Notes/Constraints 
+
+- The samples should be build inside of `./testdata/<type>/` path
```

Comment:
Just a quick doubt. Im not sure how KB does this. But, if we have different plugins for same type of operator, would we scaffold the project with default plugin version which is set.

Reply:
At the moment we have only one version of the plugin. To create the samples we use the SDK binary and run the commands at the same way that users will do that. So, in the future, if we wish we can create samples for diff plugins versions since it means `operator-sdk --plugins="plugin version"`

Reply:
@varshaprasad96 @camilamacedo86 I would NOT scaffold out samples with multiple plugins. I would create the samples for our default use case only. And possibly for any complex use cases like we are talking about with webhooks. Honestly I can see 2 samples: 1) default easy peasy project. 2) one with all the options enabled to show the complex features.

Reply:
@jmrodri it is not in the scope of the EP anyway. However, the stack allows us to do that if we wish in the future. I mean, has no limitation for that if we wish to do that. But I agree 100% with you.